webapp-django/crashstats/crashstats/views.py

```diff
@@ -1103,28 +1105,97 @@ def graphics_report(request):
+    product = form.cleaned_data['product'] or settings.DEFAULT_PRODUCT
+    date = form.cleaned_data['date']
+    params = {
+        'product': product,
+        'date': [
+            '>={}'.format(date.strftime('%Y-%m-%d')),
+            '<{}'.format(
+                (date + datetime.timedelta(days=1)).strftime('%Y-%m-%d')
+            )
+        ],
+        '_columns': (
+            'signature',
+            'uuid',
+            'date',
+            'product',
+            'version',
+            'build_id',
+            'platform',
+            'platform_version',
+            'cpu_name',
+            'cpu_info',
+            'address',
+            'uptime',
+            'topmost_filenames',
+            'reason',
+            'app_notes',
+            'release_channel',
+        ),
+        '_results_number': batch_size,
+        '_results_offset': 0,
+    }
+    api = SuperSearch()
+    # Do the first query. That'll give us the total and the first page's
+    # worth of crashes.
+    data = api.get(**params)
     assert 'hits' in data
 
     accept_gzip = 'gzip' in request.META.get('HTTP_ACCEPT_ENCODING', '')
     response = http.HttpResponse(content_type='text/csv')
     out = BytesIO()
     writer = utils.UnicodeWriter(out, delimiter='\t')
     writer.writerow(GRAPHICS_REPORT_HEADER)
-    for row in data['hits']:
-        # Each row is a dict, we want to turn it into a list of
-        # exact order as the `header` tuple above.
-        # However, because the csv writer module doesn't "understand"
-        # python's None, we'll replace those with '' to make the
-        # CSV not have the word 'None' where the data is None.
-        writer.writerow([
-            row[x] is not None and row[x] or ''
-            for x in GRAPHICS_REPORT_HEADER
-        ])
+    pages = data['total'] // batch_size
+    pages += data['total'] % batch_size and 1 or 0
+    alias = {
+        'crash_id': 'uuid',
+        'os_name': 'platform',
+        'os_version': 'platform_version',
+        'date_processed': 'date',
+        'build': 'build_id',
+        'uptime_seconds': 'uptime',
+    }
+    # Make sure that we don't have an alias for a header we don't need
+    alias_excess = set(alias.keys()) - set(GRAPHICS_REPORT_HEADER)
+    assert not alias_excess, alias_excess
+
+    def get_value(row, key):
+        value = row.get(alias.get(key, key))
+        if key == 'cpu_info':
+            value = '{cpu_name} | {cpu_info}'.format(
+                cpu_name=row.get('cpu_name', ''),
+                cpu_info=row.get('cpu_info', ''),
+            )
+        if value is None:
+            return ''
+        if key == 'date_processed':
+            value = timezone.make_aware(datetime.datetime.strptime(
+                value.split('.')[0],
+                '%Y-%m-%dT%H:%M:%S'
+            ))
+            value = value.strftime('%Y%m%d%H%M')
+        if key == 'uptime_seconds' and value == 0:
+            value = ''
+        return value
```

Comment:
I've added a doc string that explains that this whole function is just there to match what the ancient SQL based report use to be able to do.